types/node/tls.d.ts

```diff
@@ -432,6 +432,16 @@ declare module "tls" {
          * emitted with `ERR_TLS_PSK_SET_IDENTIY_HINT_FAILED` code.
          */
         pskIdentityHint?: string;
+
+        /**
+         * Indicates whether half-opened TCP connections are allowed. __Default:__ `false`.
+         */
+        allowHalfOpen?: boolean;
```

Comment:
What about extracting the options for net server into e.g. `NetServerOptions` and add it to the extends list in `TlsOptions` ?

Reply:
I'd prefer something like this. I was concerned that editing "net" definitions might be considered changing a critical package unnecessarily, but I'll be easily persuaded it's the better option.

If `NetServerOptions` sounds like a good idea, I'll make the changes and modify this PR next chance I get.

Reply:
There are already two places in net (`class Server` and `createServer`) where this new type can be used.

Looking once more into the net.d.ts file it seems that there `XXXOpts` is usually used so maybe go for `ServerOpts`. Don't think the prefix `Net` is needed as it is anyway namespaced within the net module.

Reply:
Agree - I see nine types in net.d.ts using `XXXOpts`, and just one using `XXXOptions`.